### ISSUE TITLE
[DISCUSSION] modular packages

### ISSUE THREAD

**Issue by oganexon** [nestlandofficial/nest.land#79]
In addition to the issue #67, we have to take into account another aspect.

Suppose that you want to use a small subset of a module. With the current process, you need to import everything since only `mod.ts` / a single file import is possible.

It's an issue because many packages won't be published here for this reason. Furthermore, it adds:
 -  unnecessary long compile time
 -  unnecessary huge bundle size
 - Deno packages collapsing into a black hole.

This is a little more detailed [here](https://dev.to/wongjiahau/why-deps-ts-and-mod-ts-is-bad-in-deno-bjo). Think about lodash for instance, you don't want to import this huge lib but don't want either to publish a package by feature.

So the question is:
**Do we keep this strategy, that is to say only one entry point;
Or do we rethink the package manager now, as long as there aren't many users ?**

**Comment by tbaumer22** [comment]
Thanks for starting this discussion. I should preface this with the fact that any files you include in the `files` variable in `egg.json` are accessible via the CDN. You can currently navigate to these files just by copying the relative route used in your repository and appending it to the url of the package in the CDN. 

Nest.land does not bundle these packages in any way, so importing individual modules would work perfectly fine. We simply linked to the `mod.ts` file in order to have an "index" file for users to easily import into their projects. To prove this, you can view our wallet on the Arweave ViewBlock site. Because we wanted not to alter the user's project in any way, we use an Arweave `manifest.json` file in order to map each file to each transaction on the blockchain. An example of one is [Mandarine.TS'](https://viewblock.io/arweave/tx/mtOnkDtud3fXFr3FvzF_v3hBukjINRsLKp7FbCnx36I) Make sure to click "show encoded data" when you view that.

I do agree that we don't want to mandate a `mod.ts` file, as discussed in #67. One potential solution to this would be creating an explorable file tree on the package information page. Because of the way we've built the API and CLI, this wouldn't be very difficult to implement.

**TLDR;**
We don't need to redesign anything, and I believe that we need to allow users to specify their own "index" file.

**Comment by divy-work** [comment]
:+1: Let the user optionally specify the index file at `init`

**Comment by tbaumer22** [comment]
This has been implemented! You can specify your index file in `egg.json` with the `"entry"` field now.
